src/Futhark/CLI/Autotune.hs

```diff
@@ -388,8 +414,12 @@ tune opts prog = do
   let progbin = "." </> dropExtension prog
   putStrLn $ "Running with options: " ++ unwords (serverOptions opts)
 
-  withServer (futharkServerCfg progbin (serverOptions opts)) $ \server ->
-    foldM (tuneThreshold opts server datasets) [] $ tuningPaths forest
+  fst
```

Comment:
This looks terrible; can you please use either `fmap fst . withServer` or put the `fst` inside the lambda?